bucket_test.go

```diff
@@ -464,6 +464,321 @@ func TestBucket_Delete_NonExisting(t *testing.T) {
 	}
 }
 
+func generateTestKeysValues(t testing.TB, srcBucketPath []string, db *btesting.DB) {
+	t.Helper()
+
+	var min, max = 1, 1024
+	var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	// init matrix of bucket keys
+	keys := make([][]string, 0, len(srcBucketPath))
+	values := make([][]string, 0, len(srcBucketPath))
+
+	for idx := range srcBucketPath {
+		// rand number of keys/values in each bucket
+		n := rand.Intn(max-min) + min
```

Comment:
@fuweid @ahrtr 

could this be a way to create test data ?

Reply:
For that random key/values, it looks good.

We consider to use implement a function to generate key/values with a given bucket.
What about this one?

```go
func createKeyValues(t *testing.T, bk *bbolt.Bucket, num int) {
      ...
}
```

So that, for that bucket paths, we can use that following code as example.


```go
db.Update(func(tx *bolt.Tx) error {
       for i := 0; i < len(bucketPaths); i++ {
               bk, err := createBucketIfNotExists(tx, bucketPaths[:i+1])
               if err != nil {
                      return err
               } 
               createKeyValues(t, bk, randomNum)
       }
       return nil
})

func createBucketIfNotExists(tx *bbolt.Tx, paths ...[]byte) (*bbolt.Bucket, error) {
        bk, err := tx.CreateBucketIfNotExists(paths[0])
        if err != nil {
                return nil, err
        }

        for _, key := range paths[1:] {
                bk, err = bk.CreateBucketIfNotExists(key)
                if err != nil {
                        return nil, err
                }
        }
        return bk, nil
}

```

Reply:
lgtm, but this way the `inner` buckets on the path from the `rootBucket` till the `Bucket` will not have any key/values, would this be ok ?